Model/CustomerCreditCard.php

```diff
@@ -126,4 +126,25 @@ public function recharge(OrderModel $order)
 
         return $response;
     }
+
+    /**
+     * @return object | bool
+     */
+    public function getCardInfoObject(){
+        return json_decode($this->getCardInfo(), true);
```

Comment:
I suggest ta bit refactor it to avoid using suppressions in code below.
1. Inject to the constructor additional class`Magento\Framework\Serialize\Serializer\Json $serializer`
2.  As you already injected `DataObjectFactory` the code should look like:
```
$cardInfo = $this->serializer->unserialize($this->getCardInfo());
$data = $this->dataObjectFactory->create();
$data->setData('card_info',  $cardInfo);

return $data;
```

After that we does not need to use suppressions, because:
- In object does not exist method `display_network ` it will automatically return `null`
- We should not aware about production server settings regarding suppressions
- We have predictable code, no error suppression
- We pass Magento validator because we do not have suppressions

Reply:
@YevhenShBolt @johnmccombs1 
`Magento\Framework\Serialize\Serializer\Json` is not available in M2.1.x that, I guess, we still need to support. We have active merchants on old Magento versions.
Keeping the plain old `json_decode` could be beneficial.
@matt-thomason @njfaries thoughts?

Reply:
@danac-gs @matt-thomason @njfaries - i checked and i agree with Daniel because the Magento 2.1.x do not support `Magento\Framework\Serialize\Serializer\Json` - we can easily change it to `json_decode($this->getCardInfo(), true)`

The main idea of my request is avoid suppressions and have _"clean and clear"_ look of object, instead of concatenate the object and array in one call and with suppression here.
As i see a bit above in class already used Magento's best-practice and already injected `dataObjectFactory ` to the constructor and using for work with request api object.
https://github.com/BoltApp/bolt-magento2/blob/ba9de4f0ebdb211b0ca986e917f11f2b69daa25c/Model/CustomerCreditCard.php#L101
https://github.com/BoltApp/bolt-magento2/blob/ba9de4f0ebdb211b0ca986e917f11f2b69daa25c/Model/CustomerCreditCard.php#L102

Reply:
@YevhenShBolt I updated the code. Can you review again?